File: codingame/easy/theDescent/theDescent.rb
STDOUT.sync = true # DO NOT REMOVE
# The while loop represents the game.
# Each iteration represents a turn of the game
# where you are given inputs (the heights of the mountains)
# and where you have to print an output (the index of the mountain to fire on)
# The inputs you are given are automatically updated according to your last actions.


# game loop
loop do
    i = 0
    arr = Array.new
    8.times do
        mountain_h = gets.to_i # represents the height of one mountain.
        arr[i] = mountain_h
        i += 1
    end
    
    # Write an action using puts
    # To debug: STDERR.puts "Debug messages..."

    idx = arr.index(arr.max)
    puts idx

end

File: codingame/easy/theDescent/theDescent.cs
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;

/**
 * The while loop represents the game.
 * Each iteration represents a turn of the game
 * where you are given inputs (the heights of the mountains)
 * and where you have to print an output (the index of the mountain to fire on)
 * The inputs you are given are automatically updated according to your last actions.
 **/
class Player
{
    static void Main(string[] args)
    {

        
        // game loop
        while (true)
        {
            int[] arr = new int[8];
        
            for (int i = 0; i < 8; i++)
            {
                int mountainH = int.Parse(Console.ReadLine()); // represents the height of one mountain.
                arr[i] = mountainH;
            }
            
            int answer = Array.IndexOf(arr, arr.Max());
                
            // Write an action using Console.WriteLine()
            // To debug: Console.Error.WriteLine("Debug messages...");

            Console.WriteLine(answer); // The index of the mountain to fire on.
        }
    }
}